earth2studio/models/px/dlesym.py

```diff
@@ -1060,7 +1106,89 @@ def coords_to_ll(self, coords: CoordSystem) -> CoordSystem:
             ll_coords.move_to_end(dim)
         return ll_coords
 
-    # @batch_func()
+    def _nan_interpolate_sst(
+        self, sst: torch.Tensor, coords: CoordSystem
+    ) -> torch.Tensor:
+        """
+        Custom interpolation to fill NaNs over landmasses in SST data.
+        """
+
+        da_sst = xr.DataArray(sst.cpu().numpy(), dims=coords.keys())
+        da_interp = da_sst.interpolate_na(
+            dim="lon", method="linear", use_coordinate=False
+        )
+
+        # Second pass: roll, interpolate along longitude, and unroll
+        roll_amount_lon = int(len(da_interp.lon) / 2)
+        da_double_interp = (
+            da_interp.roll(lon=roll_amount_lon, roll_coords=False)
+            .interpolate_na(dim="lon", method="linear", use_coordinate=False)
+            .roll(lon=len(da_interp.lon) - roll_amount_lon, roll_coords=False)
+        )
+
+        # Third pass do a similar roll along latitude
+        roll_amount_lat = int(len(da_double_interp.lat) / 2)
+        da_triple_interp = (
+            da_double_interp.roll(lat=roll_amount_lat, roll_coords=False)
+            .interpolate_na(dim="lat", method="linear", use_coordinate=False)
+            .roll(lat=len(da_double_interp.lat) - roll_amount_lat, roll_coords=False)
+        )
+
+        return torch.from_numpy(da_triple_interp.values).to(sst.device)
+
+    def _prepare_derived_variables(
+        self, x: torch.Tensor, coords: CoordSystem
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """
+        Prepare derived variables for the DLESyM model.
```

Comment:
Nit: First line of doc strings should start on same line as the """

E.g. """Prepare derived variables for the DLESyM model.

Reason just being this is what the autodoc extension does by default so all the package follows this

Same with _nan_interpolate_sst